tests/network/job.go

```diff
@@ -0,0 +1,51 @@
+package network
+
+import (
+	"fmt"
+	"time"
+
+	"kubevirt.io/kubevirt/tests"
+	libnetJob "kubevirt.io/kubevirt/tests/libnet/job"
+
+	batchv1 "k8s.io/api/batch/v1"
+)
+
+const (
+	toFail = false
+)
+
+// WaitForJobToFail blocks until the given job finishes.
+// On failure, it returns with a nil error, on success or timeout it returns with an error.
+func WaitForJobToFail(job *batchv1.Job, timeout time.Duration) error {
+	return libnetJob.WaitForJob(job, toFail, timeout)
+}
+
+// NewHelloWorldJobUDP takes a DNS entry or an IP and a port which it will use create a pod
+// which tries to contact the host on the provided port. It expects to receive "Hello UDP World!" to succeed.
+// Note that in case of UDP, the server will not see the connection unless something is sent over it
+// However, netcat does not work well with UDP and closes before the answer arrives, we make netcat wait until
+// the defined timeout is expired to prevent this from happening.
+func NewHelloWorldJobUDP(host, port string) *batchv1.Job {
+	timeout := 5
+	check := fmt.Sprintf(`set -x
+x=$(cat <(echo) <(sleep %[1]d) | nc -u %s %s -i %[1]d -w %[1]d | head -n 1)
+echo "$x"
+if [ "$x" = "Hello UDP World!" ]; then
+  echo "succeeded"
+  exit 0
+else
+  echo "failed"
+  exit 1
+fi`,
+		timeout, host, port)
+
+	return libnetJob.NewHelloWorldJob(check)
+}
+
+// NewHelloWorldJobHTTP gets an IP address and a port, which it uses to create a pod.
+// This pod tries to contact the host on the provided port, over HTTP.
+// On success - it expects to receive "Hello World!".
+func NewHelloWorldJobHTTP(host string, port string) *batchv1.Job {
+	check := fmt.Sprintf(`set -x; x="$(head -n 1 < <(curl --silent %s:%s))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, tests.FormatIPForURL(host), port)
```

Comment:
nit: Consider moving `FormatIPForURL` to `libnet` as well (different PR/commit).

Reply:
Done